src/server/engine_shard.cc

```diff
@@ -633,8 +633,10 @@ void EngineShard::Heartbeat() {
 void EngineShard::RetireExpiredAndEvict() {
   // TODO: iterate over all namespaces
   DbSlice& db_slice = namespaces.GetDefaultNamespace().GetDbSlice(shard_id());
-  // Some of the functions below might acquire the lock again so we need to unlock it
-  std::unique_lock lk(db_slice.GetSerializationMutex());
+  // Some of the functions below might acquire the same lock again so we need to unlock it
+  // asap. We won't yield before we relock the mutex again, so the code below is atomic
+  // in respect to preemptions.
+  { std::unique_lock lk(db_slice.GetSerializationMutex()); }
```

Comment:
1. So to understand, please comment why we acquire it - i.e. just to check none is in progress
2. We can test the atomicity with FiberAtomicGuard (just disarm it beofre TriggerJournalWriteToSink)

Reply:
>We can test the atomicity with FiberAtomicGuard (just disarm it beofre TriggerJournalWriteToSink)
I manually unlock it before, we acquired the lock in the for loop above. I will add a comment

Reply:
> I manually unlock it before, we acquired the lock in the for loop above. I will add a comment

I mean to ensure for future code changes

Reply:
What I am trying to say here, is that the `FiberGuard` won't add much because we lock the same mutex within the for loop. The bug was not that we forget to unlock it before we call TriggerJournalWriteToSink, but before it, in the for loop above.

Reply:
What am I missing in your suggestion ?